### PR TITLE
Cryp-28: list of all transactions associated to each of my eth wallets

### PR DESCRIPTION
**Pull request by alexbara2000**
Added transaction list with the ability to display date separators

### CODE REVIEW

**Comment by alexbara2000** [comment]
image
image

File: packages/client/components/Transaction.tsx [hunk 1]
```diff
@@ -22,6 +22,10 @@ export function Transaction({ transaction, walletAddress }: Props) {
         return transaction.transactionAddress.substring(0, 2) == "0x" ? "ETH" : "BTC";
     }
 
+    function formatAmount(amount: string): string {
+        return amount.length > 15 ? amount.substring(0, 15) + "..." : amount;
+    }
```
Comment (jason-gerard) [inline packages/client/components/Transaction.tsx:27]: There is a func `getFormattedAmount` in `currency_service.ts` that does this properly and then for the elipses just use the prop `isTruncated` on the `Text` tag this way its dynamic.

File: packages/client/components/TransactionList.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { Text, Box } from "native-base";
+import { StyleSheet } from "react-native";
+import { Transaction as TransactionEntity } from "@cryptify/common/src/domain/entities/transaction";
+import { Transaction } from "./Transaction";
```
Comment (jason-gerard) [inline packages/client/components/TransactionList.tsx:5]: Rename this file to a better name that doesnt conflict with the entity

```suggestion
import { TransactionListItem } from "./TransactionListItem";
```

**Review by jason-gerard — APPROVED** [review]
Logic is a bit clunky, but pretty impressed good work.

File: packages/client/components/TransactionList.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+type Props = {
+    transactions: Transaction[];
+    walletAddress: string;
+    displaySeparation: boolean;
+};
+
+export function TransactionList({ transactions, walletAddress, displaySeparation }: Props) {
+    let savedDate = new Date();
+    function renderSeparation(date: Date) {
```
Comment (domsec) [inline packages/client/components/TransactionList.tsx:15]: Just a personal preference, but something on the lines of `renderHeader` or something along those lines would be more descriptive.

File: packages/client/components/TransactionList.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                <Text color="text.500" style={styles.dateSeparator}>
+                    {date.toLocaleString("en-US", { month: "long" }) + " " + date.getFullYear()}
+                </Text>
+            </Box>
+        );
+    }
+
+    function renderTransactions(transactions: Transaction[]) {
+        return (
+            <Box backgroundColor="white" style={styles.transactionWrapper}>
```
Comment (domsec) [inline packages/client/components/TransactionList.tsx:32]: Not sure background color is needed here. Could we get away with just setting a white background to the main view and handle non white backgrounds on a per component basis? Just something to note for further investigation in the future. This works fine for now.

File: packages/client/components/TransactionList.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        );
+    }
+    return <>{renderTransactions(transactions)}</>;
+}
+
+const styles = StyleSheet.create({
+    dateSeparator: {
+        fontSize: 17,
+        fontWeight: "600",
+        marginLeft: 15,
```
Comment (domsec) [inline packages/client/components/TransactionList.tsx:63]: Should be horizontal padding of 15 and vertical padding of 5

**Review by domsec — APPROVED** [review]
We could definitely refactor this and make it more generic to support different list based screens planned for future features but for now this works. 

One minor thing I noticed based on the attached screenshots, make sure the transaction list is sorted by newest to oldest before any sorting occurs. Obviously nothing to do with this PR but I just wanted to mention it as a reminder.

Besides that, great work.

**Comment by alexbara2000** [comment]
For the order of the transactions, it is made like this by design. This component will display the transactions in the order it sees them. This is where the sorting and filtering algorithms will come in play that is currently being designed. But yes as you mention it is important we don't forget that sorting does not occur here.

File: packages/client/components/TransactionList.tsx (hunk 4 above)
Comment (alexbara2000) [reply packages/client/components/TransactionList.tsx:32]: I tried it that way originally but it kept propagating the grey color of the date separations. I will look into more details when I build my screen.

**Pull request merged**